src/status_list/mod.rs

```diff
@@ -0,0 +1,518 @@
+        }
+    }
+}
+
+impl TryFrom<StatusList> for EncodedStatusList {
+    type Error = OAuthTSLError;
+
+    fn try_from(status_list: StatusList) -> Result<Self, Self::Error> {
+        let encoded_list = status_list.compress_encode()?;
+        Ok(EncodedStatusList {
+            status_size: status_list.status_size.as_u8(),
+            status_list: encoded_list,
+            aggregation_uri: status_list.aggregation_uri,
+        })
+    }
+}
+
+impl EncodedStatusList {
+    pub fn new(status_size: Bits, status_list: String, aggregation_uri: Option<String>) -> Self {
+        EncodedStatusList {
+            status_size: status_size.as_u8(),
+            status_list,
+            aggregation_uri,
+        }
+    }
+
+    pub fn decode_decompress(&self) -> Result<Vec<u8>, OAuthTSLError> {
+        let bytes = base64::engine::general_purpose::URL_SAFE_NO_PAD.decode(&self.status_list)?;
+        let mut d = ZlibDecoder::new(&bytes[..]);
+        let mut out = Vec::new();
+        d.read_to_end(&mut out)?;
+
+        Ok(out)
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct StatusList {
+    #[serde(rename = "bits")]
+    pub status_size: Bits,
+    #[serde(rename = "lst")]
+    pub status_list: Vec<u8>,
+    // todo: not implemented yet
+    pub aggregation_uri: Option<String>,
+}
+
+/// The default Status Size is 1 bit (VALID or INVALID) and no Aggregation Uri.
+/// The default length of the status list is 50 bytes, 400 bits.
+/// This is the recommended minimum size for a status list so to encourage "herd privacy" within a Status List.
+impl Default for StatusList {
+    fn default() -> Self {
+        StatusList {
+            status_size: Bits::One,
+            status_list: vec![0u8; 50],
+            aggregation_uri: None,
+        }
+    }
+}
+
+impl TryFrom<EncodedStatusList> for StatusList {
+    type Error = OAuthTSLError;
+
+    fn try_from(encoded_list: EncodedStatusList) -> Result<Self, Self::Error> {
+        let status_list = encoded_list.decode_decompress()?;
+        Ok(StatusList {
+            status_size: Bits::try_from(encoded_list.status_size)?,
+            status_list,
+            aggregation_uri: encoded_list.aggregation_uri,
+        })
+    }
+}
+
+impl StatusList {
+    /// Takes a Vec of Statuses, each represented by a u8 and packs them into a byte according to the status size.
+    /// A status size of 8 bits would therefore need no packing.
+    /// A status size of 1 bit would pack 8 statuses into a single byte.
+    pub fn pack_statuses_into_bytes(
+        &mut self,
+        statuses: Vec<StatusType>,
+    ) -> Result<Vec<u8>, OAuthTSLError> {
+        let status_size_usize = self.status_size.as_usize();
+        let statuses_per_byte = 8 / status_size_usize;
+        let mut packed: Vec<u8> = Vec::with_capacity(statuses.len().div_ceil(statuses_per_byte));
+        let mut byte: u8 = 0;
+        let mut bit_pos = 0;
+
+        for status in statuses.iter().cloned() {
```

Comment:
`statuses` is not being used elsewhere so cloning is not needed --> `for status in statuses.into_iter()`